auth/store.go

```diff
@@ -56,10 +78,105 @@ func (as *authStore) Recover(be backend.Backend) {
 	// TODO(mitake): recovery process
 }
 
+func findUnusedBit(bitmap []byte) int {
+	for i := 0; i < len(bitmap); i++ {
+		val := bitmap[i]
+		for j := uint(0); j < 8; j++ {
+			if val&(1<<j) != 0 {
+				continue
+			}
+
+			return i*8 + int(j)
+		}
+	}
+
+	return -1
+}
+
+func (as *authStore) UserAdd(r *pb.UserAddRequest) (*pb.UserAddResponse, error) {
```

Comment:
I feel the ID generation is overly complicated. Is there a reason that we cannot just use user name as its ID? So the key in the authStore will be the username and value is the proto struct.